web/modules/custom/parser/src/Form/FilesAndTablesManagerForm.php

```diff
@@ -0,0 +1,386 @@
+<?php
+
+namespace Drupal\parser\Form;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\parser\Reader\CsvReader;
+use Drupal\parser\Reader\DiscountReader;
+use Drupal\parser\Reader\ExcelReader;
+use Drupal\parser\Reader\YamlReader;
+use Drupal\parser\Writer\DB\EntitlementsWriter;
+use Drupal\parser\Writer\DB\Rates400NGWriter;
+use Drupal\parser\Writer\DB\CsvWriter;
+use Drupal\parser\Writer\DB\DiscountWriter;
+use Drupal\parser\Writer\DB\ZipCodesWriter;
+use Drupal\Core\Url;
+use Drupal\Core\Database\Connection;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Entity\EntityTypeManager;
+
+/**
+ * Class FilesAndTablesManagerForm.
+ */
+class FilesAndTablesManagerForm extends ConfigFormBase {
```

Comment:
This class has a lot of different responsibilities, I think you need to decouple it a little bit: https://en.wikipedia.org/wiki/Single_responsibility_principle

Reply:
I think this is a better article with an example:
https://www.toptal.com/software/single-responsibility-principle
Some reasons for separate responsibilities are to have a more maintainable code, easy to read, easy to add unit test, and a better encapsulation (you can have public only the things that needs to be public, and private everything else).
If you need to add or remove code, it'd much easier to go to the proper method, instead of scanning a big file. Also, you won't mess with the already written code/files.